### ISSUE TITLE
Can't remove notifications service

### ISSUE THREAD

**Issue by sergiorodriguez82** [aws-amplify/amplify-cli#9036]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v12.16.1

### Amplify CLI Version

7.4.5

### What operating system are you using?

MAcos

### Amplify Categories

notifications

### Amplify Commands

delete, pull, push

### Describe the bug

Hi similar to: https://github.com/aws-amplify/amplify-cli/issues/2903
I cannot remove the notifications service. Someone has delete the pinpoint service, and I have a project in an uncosisnten state.
I delete manually all entries as explained in 2903, but each time I push or pull the env the notifications are "recreated".

### Expected behavior

category should be removed

### Reproduction steps

1. amplify add notifications
2. delete pinpoint manually
3. delete manually as 2903
4. amplify pull/push either would recreate the notifications category.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
│ Storage       │ tp2Storage                │ No Change │ awscloudformation │
├───────────────┼───────────────────────────┼───────────┼───────────────────┤
│ Notifications │ tp2pinpoint               │ No Change │                   │
└───────────────┴───────────────────────────┴───────────┴───────────────────┘

No changes detected
iMac-de-Sergio:trialpal2-web sergiorodriguez$ amplify notifications remove
? Choose what to remove. The Pinpoint application
? Confirm that you want to delete the associated Amazon Pinpoint application Yes
✖ Pinpoint project deletion error
Application not found
NotFoundException: Application not found
    at Object.extractError (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/protocol/json.js:52:27)
    at Request.extractError (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/protocol/rest_json.js:49:8)
    at Request.callListeners (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/Users/sergiorodriguez/.nvm/versions/node/v12.16.1/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/event_listeners.js:335:13)
    at IncomingMessage.emit (events.js:323:22)
    at IncomingMessage.EventEmitter.emit (domain.js:482:12)
    at endReadableNT (_stream_readable.js:1204:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
iMac-de-Sergio:trialpal2-web sergiorodriguez$ 


```

</details>

**Comment by kaustavghosh06** [comment]
@sergiorodriguez82 If the app is deleted manually from the console, you can remove references of it from your amplify project by editing  the `amplify/backend/backend-config.json` file as well as `amplify/backend/amplify-meta.json`. The references of the resource would be under the "notifications" key in these files.

**Comment by sergiorodriguez82** [comment]
I have done that, but when pushing changes the notifications references appear again

**Comment by lazpavel** [comment]
Hi @sergiorodriguez82, making changes to the project outside CLI is not supported. Can you try running `amplify remove notifications` and let us know if this solves the problem.

**Comment by sergiorodriguez82** [comment]
When doing the remove command, it throws an exception, we decided to create a new environment, and deleting the one with the problem, nevertheless if this was a production environment we would have a serious problem. I understand if something is done outside the amplify/cloudformation is hard to sync the model. However, this is something that can happen, and I consider there must be a way to get the environment in a consistent state.
Thanks!

**Comment by josefaidt** [comment]
Hey @sergiorodriguez82 apologies for the delay! I've been able to successfully reproduce this issue using the steps provided. As noted I deleted the pinpoint project manually from the console and was then unable to remove via the CLI with `amplify remove notification`, however I did not experience the issue where the notification project was recreated on subsequent push or pull.

Using the manual workarounds we are able to remove the category from our project locally:

1. remove all references to `notifications` from:
	1. `amplify/team-provider-info.json`
	2. `amplify/backend/amplify-meta.json`
	3. `amplify/backend/backend-config.json`
2. run `amplify status` and observe notifications category now shows status `Delete`
3. run `amplify push --force` to complete the "deletion"

Marking this as a bug 🙂 I think ideally if the resource was deleted from the console we should also be able to successfully run `amplify remove notifications` to remove the resource from the project locally.
